### PR TITLE
Python dependencies only

### PR DESCRIPTION
**Pull request by pnhofmann**

### CODE REVIEW

**Comment by pnhofmann** [comment]
Patches for #336. Please take a look.
Also includes new remind-command #339 - since old remind command needs dbus.

Please take a look! Especially - since I can't do that - someone test "remind" on MacOS - and check if the notification works as expected.

**Comment by StefanosChaliasos** [comment]
@pnhofmann Did you check all the tests? If you want help with tests you can open an issue.

**Comment by pnhofmann** [comment]
@StefanosChaliasos 
Thanks!
Fixed music test and some code still issues.
The timeIn Test failure has nothing to do with these patches - in fact this test case even fails on with "beforepluginmanager". Created issue #342

**Comment by appi147** [comment]
Is it ready @pnhofmann ?

**Comment by pnhofmann** [comment]
@appi147 
Mostly. In fact, I have only some thought about setup.sh

The setup.sh on Mac writes 'python' or 'python3' to jarvis depending if python2 or python3 is used. Is this really necessary - since on Linux with python3 virtualenv python is symbolic link to python3. Does virtualenv really differ between Mac and Linux? Because if not, we could remove the whole os-specific code in setup.sh and just check in
```
#!/bin/bash
source /home/phil/projects/Jarvis/env/bin/activate
python /home/phil/projects/Jarvis/jarviscli/
```
as jarvis.
 
@StefanosChaliasos 
About
``` cp jarvis /usr/local/bin```
On most systems we would probably need ``sudo``. Don't really like doing anything outside /home in setup.sh.

I would propose asking the user:

```
echo "How to start Jarvis?"
echo "1: Call full path $jarvispath/jarvis"
echo "2: Copy jarvis to /usr/local/bin (root required)"
echo "3: Add $jarvispath to \$PATH"
read -n 1 -p "(Default: 1) " answer
echo ""
echo ""
echo "Installation complete. Try using Jarvis!"
case ${answer:0:1} in
2 )
    sudo cp jarvis /usr/local/bin
    echo "\$ jarvis"
;;
3)
    echo "export PATH=\$PATH:$jarvispath" >> ~/.bashrc
    echo "\$ jarvis"
;;
* )
    echo "\$ $jarvispath/jarvis"
;;
esac
```

**Comment by appi147** [comment]
Don't know about  MacOS, but it would be good. MacOS comes with python2 as default python IMO(depends on version of MacOS). Adding $jarvispath to path would be a good option. Don't know if we can set path same in MacOS.

**Comment by appi147** [comment]
OK here are tests:
- remind works(sent a small notification)
- remind remove crashes Jarvis if there is no reminder(empty reminder list)
- facebook and twitter are not present(as seen in commits)
- sound is fine
- hackathon is fine
Python2 is fine

**Comment by pnhofmann** [comment]
@appi147
Thanks! Fixed reminder crash.

**Comment by pnhofmann** [comment]
Ok if I rebase master or is a merge prefered?

**Comment by appi147** [comment]
Rebase pls

File: setup.sh [hunk 1]
```diff
@@ -1,4 +1,18 @@
 #!/bin/bash
+# fail on error!
+set -euo pipefail
```
Comment (appi147) [inline setup.sh]: [https://github.com/pypa/virtualenv/issues/1029](https://github.com/pypa/virtualenv/issues/1029)
`source /env/bin/activate` fails on my machine

File: setup.sh [hunk 2]
```diff
@@ -40,98 +54,20 @@ exec 3<> jarvis
...
   ;;
 esac
 
 source env/bin/activate
-pip install -r requirements.txt
-pip install dbus-python
+
+pip install --upgrade -r requirements.txt
+
+cp jarvis /usr/local/bin
```
Comment (appi147) [inline setup.sh]:
```suggestion
sudo cp jarvis /usr/local/bin
```

**Comment by appi147** [comment]
@pnhofmann

**Comment by appi147** [comment]
Excellent work @pnhofmann

**Pull request merged**